tests/test_ts_restore_malformed.py

```
@@ -0,0 +1,116 @@
+        start = p + m.end()
+        payload = data[start:start + int(m.group(1))]
+        assert len(payload) > 0
+        return payload
+
+    def test_restore_garbage_payload_does_not_crash(self):
+        """Payloads that are not module output at all are rejected, not fatal."""
+        client = self.client
+        for i, blob in enumerate([b'', b'garbage', b'\x01\x02\x03', b'A' * 32, b'\x00' * 256]):
+            try:
+                client.execute_command('TS._RESTORE', f'garbage{i}', blob)
+            except Exception as e:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)